src/server/game/Movement/MovementGenerators/FollowMovementGenerator.cpp

```diff
@@ -53,12 +53,18 @@ inline UnitMoveType SelectSpeedType(uint32 moveFlags)
     return MOVE_RUN;
 }
 
-inline bool IsTargetMoving(Unit const* target)
+static bool IsTargetMoving(Unit const* owner, Unit const* target, float distance)
 {
-    return target->HasUnitMovementFlag(MOVEMENTFLAG_FORWARD | MOVEMENTFLAG_BACKWARD | MOVEMENTFLAG_STRAFE_LEFT | MOVEMENTFLAG_STRAFE_RIGHT) || !target->movespline->Finalized();
+    if (target->HasUnitMovementFlag(MOVEMENTFLAG_FORWARD | MOVEMENTFLAG_BACKWARD | MOVEMENTFLAG_STRAFE_LEFT | MOVEMENTFLAG_STRAFE_RIGHT) || !target->movespline->Finalized())
+        return true;
+
+    if (owner->GetExactDistSq(target) > square(owner->GetCombatReach() + target->GetCombatReach() + distance))
```

Comment:
this does not belong into this helper. Furthermore this also causes following npcs to prematurely interrupt their follow movement when a target gets within the hitbox range which is wrong. We do not care about the combat reach here at all and strictly rely on the radius that we have set in the first place.

Reply:
I will do related changes after wake up. Can make another helped called "IsWithinDistRadius"

Reply:
I applied changes actually, should be better now :)